python/pyarrow/lib.pxd

```diff
@@ -611,3 +623,6 @@ cdef public shared_ptr[CTensor] pyarrow_unwrap_tensor(object tensor)
 
 cdef public shared_ptr[CRecordBatch] pyarrow_unwrap_batch(object batch)
 cdef public shared_ptr[CTable] pyarrow_unwrap_table(object table)
+
+cdef public CFunctionRegistry* pyarrow_unwrap_function_registry(object registry)
```

Comment:
should we consider using a `shared_ptr` instead? 
cc @lidavidm

Reply:
This has crossed my mind when I was working on the code of this PR. I also think a `shared_ptr` would make things easier but will require changes across Py/Arrow that are likely better handled in a dedicated PR.